### PR TITLE
Make Confal pyramids colors changeable

### PR DESCRIPTION
**Pull request by MadCatX**
Let's use proper Molstar facilities for confal pyramids color theme

### CODE REVIEW

**Comment by arose** [comment]
@MadCatX could you use a `default` and `custom` approach for the changeable colors like here

https://github.com/molstar/molstar/blob/6e5c20f442af01a85aff27e2bad956280532630d/src/mol-theme/color/element-symbol.ts#L42-L45

That way the colors are only stored in the state when the `custom` option is used.

**Comment by MadCatX** [comment]
> @MadCatX could you use a `default` and `custom` approach for the changeable colors like here
> 
> https://github.com/molstar/molstar/blob/6e5c20f442af01a85aff27e2bad956280532630d/src/mol-theme/color/element-symbol.ts#L42-L45
> 
> That way the colors are only stored in the state when the `custom` option is used.

Nice, that's very sensible. Fixed in the updated code.

File: src/extensions/dnatco/confal-pyramids/color.ts [hunk 1]
```diff
@@ -11,149 +11,248 @@ import { Location } from '../../../mol-model/location';
...
-    ['ZZ2S', ['Z', 'SYN']],
-    ['ZZS1', ['SYN', 'Z']],
-    ['ZZS2', ['SYN', 'Z']],
-]);
+const PyramidsColors = ColorMap({
+    NANT_Upr: DefaultClassColors.N,
+    NANT_Lwr: DefaultClassColors.N,
+    AA00_Upr: DefaultClassColors.A,
+    AA00_Lwr: DefaultClassColors.A,
+    AA02_Upr: DefaultClassColors.A,
```
Comment (dsehnal) [inline src/extensions/dnatco/confal-pyramids/color.ts:40]: To reduce the amount of code here, you can do:

```ts
{
  N: ['NANT_Upr', 'NANT_Lwr', ...],
  A: [...]
}
```

and transforms that to the object accepted by the ``ColorMap``.

Comment (MadCatX) [reply src/extensions/dnatco/confal-pyramids/color.ts]: Thanks for the suggestion. I tried it out to see what it'd look like but I'm not sure how I feel about it. The original object - while quite verbose - always kept the upper and lower part of the pyramid together whereas this compact mapping spreads them across multiple arrays. My other concern is that the original code created a logically laid out legend table that followed the published NtC ordering, this compact code doesn't.

Orignal (upper and lower parts are always together):
mapping_original

Compact ("random-ish" order from the user's perspective):
mapping_compact

Comment (dsehnal) [reply src/extensions/dnatco/confal-pyramids/color.ts]: Fair enough, thanks for trying it. I guess it's ok to reverse it back then.

**Review by dsehnal — APPROVED** [review]
LGTM, thanks for the PR.

**Pull request merged**